app/org/sagebionetworks/bridge/dynamodb/DynamoCompoundActivityDefinition.java

```diff
@@ -0,0 +1,116 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.util.List;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverted;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.ImmutableList;
+
+import org.sagebionetworks.bridge.models.schedules.CompoundActivityDefinition;
+import org.sagebionetworks.bridge.models.schedules.SchemaReference;
+import org.sagebionetworks.bridge.models.schedules.SurveyReference;
+
+/**
+ * DynamoDB implementation of CompoundActivityDefinition. The hash key is study ID and the range key is the task ID.
+ * This allows us to namespace compound activity definitions to studies and to list all definitions in a study.
+ */
+@DynamoDBTable(tableName = "CompoundActivityDefinition")
+@JsonFilter("filter")
+public class DynamoCompoundActivityDefinition implements CompoundActivityDefinition {
+    private List<SchemaReference> schemaList = ImmutableList.of();
+    private String studyId;
+    private List<SurveyReference> surveyList = ImmutableList.of();
+    private String taskId;
+    private Long version;
```

Comment:
The task ID is the same task ID as in TaskReference and in the Study.getTaskIdentifiers(). My assumption was that we didn't need a new set of identifiers for Compound Activities since the idea was to associate a list of schemas and surveys to tasks, and that there would always be a one-to-one mapping between task IDs and those lists. We _could_ add a separate set of CompoundActivityDefinitionIds, but I feel like this would only add confusion, and I'm not sure what the benefits are.

The task IDs themselves are already defined by researchers/developers in the Bridge Study Manager, so I assume they're already human readable. The assumption is whatever we currently use for TaskReferences could also be used for CompoundActivities. If for whatever that user experience is still confusing, I could trivially add name to CompoundActivity and CompoundActivityDefinition.